wweval/R/combine_and_summarize_ww_data.R

```diff
@@ -0,0 +1,417 @@
+#' dataset for a particular forecast date and location. It computes summary
+#' statistics on the wastewater characteristics for that state on that
+#' forecast date. It will return a single row dataframe with the relevant
+#' metadata.
+#'
+#'
+#' @param fp_hosp Character string indicating the filepath for the hospital
+#' admissions dataset, expected as a tsv
+#' @param fp_ww Character string indicating the filepath for the hospital
+#' admisisons dataset, expected as a tsv
+#' @param this_forecast_date Character string indicating the forecast date in
+#'  ISO8601 format (YYYY-MM-DD)
+#' @param this_location Character string indicating location, formatted as
+#' state abbreviation
+#'
+#' @return this_ww_metadata: a 1 row dataframe with metadata on the wastewater
+#' summary statistics for that forecast date and location
+#' @export
+load_data_and_summarize <- function(fp_hosp, fp_ww,
+                                    this_forecast_date,
+                                    this_location) {
+  # Use hospital data to get state pop
+  this_hosp_data <- readr::read_tsv(fp_hosp)
+  state_pop <- this_hosp_data |>
+    dplyr::distinct(.data$pop) |>
+    dplyr::pull()
+  if (length(state_pop) != 1) {
+    cli::cli_abort(message = "multiple state pops reported")
+  }
+
+  this_ww_data <- readr::read_tsv(fp_ww)
+  if (!nrow(this_ww_data) == 0) {
+    n_sites <- this_ww_data |>
+      dplyr::distinct(.data$site) |>
+      nrow()
+    n_labs <- this_ww_data |>
+      dplyr::distinct(.data$lab) |>
+      nrow()
+    sum_site_pops <- this_ww_data |>
+      dplyr::group_by(.data$site) |>
+      dplyr::summarise(
+        mean_site_pop = mean(.data$ww_pop, na.rm = TRUE)
+      ) |>
+      dplyr::ungroup() |>
+      dplyr::summarise(ww_total_pop = sum(.data$mean_site_pop,
+        na.rm = TRUE
+      )) |>
+      dplyr::pull(.data$ww_total_pop)
+    pop_coverage <- sum_site_pops / state_pop
+
+    avg_latency <- this_ww_data |>
+      dplyr::group_by(.data$lab_wwtp_unique_id) |>
+      dplyr::summarize(max_date = max(.data$date)) |>
+      dplyr::mutate(
+        latency = as.numeric(lubridate::ymd(
+          !!this_forecast_date
+        ) - lubridate::ymd(.data$max_date))
+      ) |>
+      dplyr::summarize(
+        mean_latency = mean(.data$latency, na.rm = TRUE)
+      ) |>
+      dplyr::pull(.data$mean_latency)
+
+    avg_sampling_freq <- this_ww_data |>
+      dplyr::group_by(.data$lab_wwtp_unique_id) |>
+      dplyr::arrange(.data$date, desc = TRUE) |>
+      # There are some duplicate dates within a site and lab
+      dplyr::distinct(.data$date) |>
+      dplyr::mutate(
+        prev_date = dplyr::lag(.data$date, 1),
+        diff_time = as.numeric(lubridate::days(
+          difftime(.data$date, prev_date)
+        ), "days")
+      ) |>
+      dplyr::ungroup() |>
+      dplyr::summarize(
+        mean_collection_freq = mean(.data$diff_time, na.rm = TRUE)
+      ) |>
+      dplyr::pull(.data$mean_collection_freq)
```

Comment:
This isn't exactly the mean frequency. It's the mean gap between samples until the last. Simpler and closer to what we want to compute `n_days_with_samples / n_days_total`, no?

To see why these are different, consider an extreme case: a site with only two observed days in our 28 day period, day 1 and day 2. Its mean "frequency" on this metric is going to be 1, even though only one out of 14 days (and none of the final 26) were sampled. It will also give the result `NaN` when there's only one date, since the `prev_date` will be `NA`.

```R
single_date_df <- tibble::tibble(date="2024-01-01")

single_date_df |>
dplyr::mutate(
        prev_date = dplyr::lag(.data$date, 1),
        diff_time = as.numeric(lubridate::days(
          difftime(.data$date, prev_date)
        ), "days")
      ) |>
      dplyr::ungroup() |>
      dplyr::summarize(
        mean_collection_freq = mean(.data$diff_time, na.rm = TRUE)
      ) |>
      dplyr::pull(.data$mean_collection_freq)
```

Reply:
Yeah the word frequency isn't correct here -- its the average days between samples but will give NA if only 1. Also number of days with samples and days total are probably themselves useful metrics. Will add.

Reply:
If you're interested in how big typical gaps are in the data I think still need a different metric. Can discuss.

Reply:
I recomputed as the mean over the sites of (`n_days_w_samples/n_total_days`)

Reply:
The reason being that if you use the whole states data across all sites, this becomes meaningless in states with a large number of sites (they will nearly always have n_days_w_samples = n_total_days so I'm reasoning that this is giving you more information. Also later we average across states and forecast dates in this same way